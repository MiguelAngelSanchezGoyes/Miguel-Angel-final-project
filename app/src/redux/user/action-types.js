export const userActionTypes = {
  load: '@user/load',
  add: '@user/add',
  login: '@user/login',
  logout: '@user/logout',
  // verify: '@user/verify',
};
