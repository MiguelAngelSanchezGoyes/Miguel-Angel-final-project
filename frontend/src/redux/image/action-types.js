export const imageActionTypes = {
  load: '@image/load',
  add: '@image/add',
  update: '@image/update',
  remove: '@image/remove',
};
