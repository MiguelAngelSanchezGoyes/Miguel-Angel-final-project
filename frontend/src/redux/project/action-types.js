export const projectActionTypes = {
  load: '@project/load',
  loadById: '@project/loadById',
  add: '@project/add',
  edit: '@project/edit',
  remove: '@project/remove',
};
