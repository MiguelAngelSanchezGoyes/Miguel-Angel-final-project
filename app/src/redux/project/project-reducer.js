/* eslint-disable no-underscore-dangle */
import { projectActionTypes } from './action-types';

const initialProjectState = {
  projects: [],
};

const projectReducer = (state = initialProjectState, action) => {
  switch (action.type) {
    case projectActionTypes.load:
      return { ...state, projects: [...action.projects] };
    case projectActionTypes.add:
      return { ...state, projects: [...state.projects, action.project] };
    case projectActionTypes.remove:
      return {
        ...state,
        projects: state.projects.filter((item) => item._id !== action.id),
      };
    default:
      return state;
  }
};

export default projectReducer;
