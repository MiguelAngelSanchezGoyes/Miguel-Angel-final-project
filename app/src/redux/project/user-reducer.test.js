import { projectReducer } from './project-reducer';
import { projectActionTypes } from './action-types';

describe('userReducer reducer', () => {
  const initialState = [];
  test('should return the initial state', () => {
    expect(projectReducer(undefined, [])).toMatchSnapshot();
  });

  test('should handle projectReducer.add', () => {
    expect(
      projectReducer(initialState, {
        type: projectActionTypes.add,
      })
    ).toMatchSnapshot();
  });

  test('should handle projectReducer.load', () => {
    expect(
      projectReducer(initialState, {
        type: projectActionTypes.load,
      })
    ).toMatchSnapshot();
  });

  test('should handle projectReducer.update', () => {
    expect(
      projectReducer(initialState, {
        type: projectActionTypes.update,
      })
    ).toMatchSnapshot();
  });

  test('should handle projectReducer.remove', () => {
    expect(
      projectReducer(initialState, {
        type: projectActionTypes.remove,
      })
    ).toMatchSnapshot();
  });
});
