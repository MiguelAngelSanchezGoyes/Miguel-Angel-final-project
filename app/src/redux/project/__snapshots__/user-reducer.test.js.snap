// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userReducer reducer should handle projectReducer.update 1`] = `Array []`;

exports[`userReducer reducer should return the initial state 1`] = `
Object {
  "projects": Array [],
}
`;
